kube/objects.go

```diff
@@ -159,7 +193,79 @@ func (c *Client) FetchObjects(ctx context.Context, filter ObjectFilter) (*Object
 		return nil, err
 	}
 
-	return objects, nil
+	return objectsWithoutNils(objects), nil
+}
+
+func annotateFetchError(kind string, err error) error {
+	if err == nil {
+		return nil
+	}
+	if kerrors.IsNotFound(err) {
+		// Resource doesn't exist in this cluster's version, so there aren't any
+		// objects to list and check.
+		return nil
+	}
+
+	return fmt.Errorf("failed to fetch %s: %s", kind, err)
+}
+
+func objectsWithoutNils(objects *Objects) *Objects {
```

Comment:
This now becomes a bit of a maintenance burden since we always have to remember to extend the function when a new Kubernetes resource is supported.

As an alternative suggestion, we could move the initialization of the non-nil `*Lists` into a new `NewObjects()` factory method and then adjust the individual goroutines in `FetchObjects()` like this:

```go
g.Go(func() error {
    objs, err := client.PersistentVolumes().List(gCtx, opts)
    if aerr := annotateFetchError("PersistentVolumes", err); aerr != nil {
        return aerr
    }
    objects.PersistentVolumes = objs
    return nil
})
```

It's a bit more verbose, but to me it feels more natural to work with.

Reply:
Nevermind, the above doesn't work because we need to distinguish between no errors at all, ignorable errors where we do should not set the objects (not found), and true errors. This will blow up the code by quite a bit, so I think your original approach has the better trade-offs.